### DISCUSSION TITLE
check if a point is inside a polygon

### DISCUSSION THREAD

**Discussion by amkfat** [clickbar/laravel-magellan#43] [Q&A] [answered]
I am trying to check if a Point is inside a Polygon but I'm always getting false . here is my code

Port.php

`
class Port extends Model
{
    use HasPostgisColumns;

    protected $fillable = [
        'name',
        'country',
        'location',
    ];

    protected array $postgisColumns = [
        'location' => [
            'type' => 'geometry',
            'srid' => 4326,
        ],
    ];
`

the controller:

`   $point1 = Point::makeGeodetic(100, 0);
     $point2 = Point::makeGeodetic(101, 0);
     $point3 = Point::makeGeodetic(101, 1);
     $point4 = Point::makeGeodetic(100, 1);

     $lineString = LineString::make([$point1, $point2, $point3, $point4, $point1]);

     $polygon = Polygon::make([$lineString]);`
     dump($ports);

     $ports = Port::select('location')
            ->stSelect(ST::contains($polygon2, 'location'))
            ->get();

      dd($ports);

`

the output:
<img width="866" alt="Screenshot 2023-06-24 at 11 13 29" src="https://github.com/clickbar/laravel-magellan/assets/76127580/6d9fe4bb-cd0f-4101-a00a-da2d06032388">

<img width="743" alt="Screenshot 2023-06-24 at 11 14 23" src="https://github.com/clickbar/laravel-magellan/assets/76127580/17e4c448-ee40-4ba3-b497-4acd6d6c9af5">

i checked on 'geojson .io', the point is inside the polygon:

<img width="1177" alt="Screenshot 2023-06-24 at 11 07 22" src="https://github.com/clickbar/laravel-magellan/assets/76127580/b21941d0-fc9e-4e48-8e1e-d0d592ff56d8">

**Comment by saibotk** [comment]
It seems like your Port location point's coordinates have the x and y mixed up.

**Reply by amkfat** [reply]
thanks.

**Comment by ahawlitschek** [comment, marked as answer]
Short further explanation: 

If you look into the following readme [section](https://github.com/clickbar/laravel-magellan#using-the-geometry-data-classes) you can see the differences when creating points. 

There is the normal `Point::make(...)` function, which accepts x,y,z and then the `Point::makeGeodetic(...)` which accepts latitude, longitude. 

We added the makeGeodetic function, because a lot of people will use WGS84 as Projection. And we wanted to input to be latitude, longitude instead of longitude, latitude.

Remember: 
Latitude -> y
Longitude -> x

**Reply by amkfat** [reply]
it works now.
thanks.
